### PR TITLE
Rewrite pipeline2ATX algorithm

### LINKED ISSUE #26: Rewrite pipeline2ATX
**Describe the feature request**

As a maintainer I want to have a better to maintain and flexible implementation of the pipeline2ATX function

**Upstream changes**

- [x] pipeline2ATX.groovy

**Acceptence criteria**

- [x] interface stays the same
- [x] stages are the top-level of the report (like in Jenkins)
- [x] all test steps are reported as test steps
- [x] edge case handling is well documented in the code

### PR DESCRIPTION
**Pull request by ErikRehmTT**
Resolves #26

### CODE REVIEW

**Comment by MartinGroscheTT** [comment]
@ErikRehmTT please fix tests first and add some more if needed

File: vars/pipeline2ATX.groovy [hunk 1]
```diff
@@ -199,22 +204,22 @@ static def resultToATXVerdict(result) {
...
 
     // Table might not exist before execution is finished
     FlowGraphTable table = new FlowGraphTable(build.getExecution())
     table.build()
 
     // Get start of relevant pipeline steps
-    def topLevelRows = table.getRows().stream().findAll { r -> r.getTreeDepth() == 3 }.toArray()
-    for (FlowGraphTable.Row row : topLevelRows) {
-        def item = crawlRows(row, debug)
+    def Stages = table.getRows().stream().findAll { r -> r.getDisplayName() == "stage" }.toArray()
```
Comment (MartinGroscheTT) [inline vars/pipeline2ATX.groovy:220]: I guess this finds all stages. TopLevel stages and nested ones?

**Review by MartinGroscheTT — CHANGES_REQUESTED** [review]

File: vars/pipeline2ATX.groovy (hunk 1 above)
Comment (ErikRehmTT) [reply vars/pipeline2ATX.groovy:220]: Yes, that is the case. So we get a similar view like in Jenkins having all stages as top-level execution steps.

**Review by MartinGroscheTT — APPROVED** [review]
